services/kvstoremanager/kvstoremanager.go

```diff
@@ -49,3 +56,21 @@ func EventToKeyValue(jsonData json.RawMessage) (string, map[string]interface{})
 
 	return key, fields
 }
+
+// IsHSETCompatibleEvent identifies if the event supports HSET operation
+// To support HSET, the event must have the following fields:
+// - message.key
+// - message.value
+// - message.hash
+// It doesn't account for the value of the fields.
+func IsHSETCompatibleEvent(jsonData json.RawMessage) bool {
+	return gjson.GetBytes(jsonData, hashPath).Exists() && gjson.GetBytes(jsonData, keyPath).Exists() && gjson.GetBytes(jsonData, valuePath).Exists()
```

Comment:
I would avoid the extensive and repeated usage of `gjson.` 

Using a go struct 
```go 
type HSetEvent struct {
   Hash string `json:"<hash_path>"`
   Key string `json:"<key_path>"`
   Value string `json:"<value_path>"`
}
```
It would be safer and faster (assuming fastjson library is used)

Reply:
I am not sure if I understand `faster (assuming fastjson library is used)` completely.

I can use fastjson's [Exists](https://pkg.go.dev/github.com/valyala/fastjson#Exists) but that doesn't require a go struct to unmarshal to.
Unless we somehow use fastjson while unmarshalling, using CustomUnmarshal, or I might be completely missing the point here.